File: BasicClassValidation/Classes/Taxpayer.vb
Imports System.ComponentModel.DataAnnotations

Namespace Classes
    Public Class Taxpayer
        <RegularExpression("^\d{9}|\d{3}-\d{2}-\d{4}$", ErrorMessage:="Invalid Social Security Number")>
        <Required(ErrorMessage:="Contact {0} is required"), DataType(DataType.Text)>
        Public Property SSN() As String
        <Required(ErrorMessage:="{0} is required"), DataType(DataType.Text)>
        Public Property FirstName() As String
        <Required(ErrorMessage:="{0} is required"), DataType(DataType.Text)>
        Public Property LastName() As String
    End Class
End Namespace

File: SimpleValidation/Classes/Taxpayer.cs
using System.ComponentModel.DataAnnotations;
using WindowsForms.DataAnnotationsValidation;

namespace SimpleValidation.Classes
{
    public class Taxpayer : BaseModel
    {

        [RegularExpression("^\\d{9}|\\d{3}-\\d{2}-\\d{4}$", ErrorMessage = "Invalid Social Security Number")]
        [Required(ErrorMessage = "{0} is required"), DataType(DataType.Text)]
        public string SSN { get; set; }

        [Required(ErrorMessage = "{0} is required"), DataType(DataType.Text)]
        [StringLength(20, MinimumLength = 3, ErrorMessage = "Invalid {0}")]
        public string FirstName { get; set; }

        [Required(ErrorMessage = "{0} is required"), DataType(DataType.Text)]
        [StringLength(30, MinimumLength = 5, ErrorMessage = "Invalid {0}")]
        public string LastName { get; set; }
    }
}